src/PlantBundle/Controller/PlantController.php

```diff
@@ -21,5 +22,45 @@ public function indexAction()
 
         return $this->render('Plant/index.html.twig', ['plants' => $plants]);
     }
+    /**
+     * @Route("/add-{id}", name="addPlant")
+     */
+    public function addAction($id)
+    {
+        $user=$this->getUser();
```

Comment:
`$this->getUser()` can throw an exception if user does not exist (e.g. anonymous user went to that URI). you should check if `$user` is not null